File: app/src/main/java/fr/free/nrw/commons/mwapi/UserInterface.java
package fr.free.nrw.commons.mwapi;

import static org.wikipedia.dataclient.Service.MW_API_PREFIX;

import io.reactivex.Observable;
import org.wikipedia.dataclient.mwapi.MwQueryResponse;
import retrofit2.http.GET;

public interface UserInterface {

    /**
     * Checks to see if a user is currently blocked from Commons
     */
    @GET(MW_API_PREFIX + "action=query&meta=userinfo&uiprop=blockinfo")
    Observable<MwQueryResponse> getUserBlockInfo();
}

File: app/src/main/java/fr/free/nrw/commons/mwapi/UserInterface.kt
package fr.free.nrw.commons.mwapi

import fr.free.nrw.commons.wikidata.WikidataConstants
import fr.free.nrw.commons.wikidata.mwapi.MwQueryResponse
import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Query
import retrofit2.http.QueryMap

interface UserInterface {
    /**
     * Gets the log events of user
     * @param user name of user without prefix
     * @param continuation continuation params returned in previous query
     * @return query response
     */
    @GET(WikidataConstants.MW_API_PREFIX + "action=query&list=logevents&letype=upload&leprop=title|timestamp|ids&lelimit=500")
    fun getUserLogEvents(
        @Query("leuser") user: String?,
        @QueryMap continuation: Map<String?, String?>?
    ): Observable<MwQueryResponse>

    /**
     * Checks to see if a user is currently blocked from Commons
     */
    @GET(WikidataConstants.MW_API_PREFIX + "action=query&meta=userinfo&uiprop=blockinfo")
    fun getUserBlockInfo(): Observable<MwQueryResponse>
}
